node/builder.go

```diff
@@ -566,6 +566,13 @@ func ConfigStorageMiner(c interface{}) Option {
 		return Error(xerrors.Errorf("invalid config from repo, got: %T", c))
 	}
 
+	pricingConfig := cfg.Dealmaking.RetrievalPricing
+	if pricingConfig.Strategy == config.ExternalRetrievalPricing {
+		if pricingConfig.External == nil || pricingConfig.External.Path == "" {
+			return Error(xerrors.New("invalid external retrieval pricing config"))
```

Comment:
Let's separate out the error messages for External == nil and the path being the empty string - probably users are going to make these mistakes so we want to give them as much information as possible about what is wrong so they can fix it